pkg/adaptation/default-validator.go

```diff
@@ -0,0 +1,173 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package adaptation
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/containerd/nri/pkg/log"
+	yaml "gopkg.in/yaml.v3"
+)
+
+type DefaultValidatorConfig struct {
+	Enable         bool `toml:"enable"`           // enable default validator
+	RejectOCIHooks bool `toml:"reject_oci_hooks"` // reject OCI hook injection
```

Comment:
Should we also give the option to disable validating that required plugins are present?

Reply:
Would that be useful ? I mean, this is already by definition an opt-in feature since you need to actively annotate your containers/pods for the 'processed by required plugins' check to take place. Wouldn't adding a separate configuration knob make this more error prone to use ? Now it's enough to ensure that validation is enabled and then opt-in for this check by annotating. With an extra knob, you'd also need to make sure that the extra knob is in the right position.

However, now that you asked about enabling/disabling, I think one thing that we should consider is at least logging a warning or an error, if the related annotation is present but we are running with disabled validation.

Reply:
I think yes.

If we configure required plugins also as part of the container runtime configuration, an annotation that allows the required plugins to be bypassed would allow those required plugins to be deployed as containers themselves.